### DISCUSSION TITLE
Support specifying an alternative gravatar-compatible address

### DISCUSSION THREAD

**Discussion by nogweii** [grafana/grafana#36993] [General]
- What Grafana version are you using? **4.6.2**
- What OS are you running grafana on? **Ubuntu 16.04**

I like avatars to help personalize the user experience, but I don't want to use Gravatar's service to query them for the internal grafana dashboard at my company. I don't expect Grafana to implement a whole pluggable API just for avatars, just that it'd be nice to be able to override the `srcCfg` variable with something from the configuration file.

https://github.com/grafana/grafana/blob/a4972f44adf1e893a4b6b5b435728ad70a48c43c/pkg/api/avatar/avatar.go#L34-L44

In the mean time, I think I could configure nginx to intercept `/avatar/MD5` and send it off somewhere else:

https://github.com/grafana/grafana/blob/af34f9977ed894fbe9abb2ce02ebff872f194b3c/pkg/api/dtos/models.go#L52-L60

**Comment by Luckyvb** [comment]
for nginx all ok:
```
    location /avatar/ {
        root /var/www/html;
        try_files $uri.png $uri.png @backend;
        sendfile  on;
    }
   location @backend {
       proxy_pass http://localhost:3000; #as sample
       ...
   }
```
if no nginx, need changes, like you wrote

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #83005. Feel free to continue the discussion around this there. Thank you!
